fe/fe-core/src/main/java/com/starrocks/catalog/CatalogRecycleBin.java

```diff
@@ -217,22 +229,50 @@ public synchronized List<Partition> getPartitions(long tableId) {
                 .collect(Collectors.toList());
     }
 
-    private synchronized boolean isExpire(long id, long currentTimeMs) {
+    /**
+     * if we can erase this instance, we should check if anyone enable erase later.
+     * Only used by main loop.
+     */
+    private synchronized boolean canErase(long id, long currentTimeMs) {
         long latency = currentTimeMs - idToRecycleTime.get(id);
+        if (enableEraseLater.contains(id)) {
+            latency -= LATE_RECYCLE_INTERVAL_SECONDS * 1000L;
```

Comment:
Why to minus the LATE_RECYCLE_INTERVAL_SECONDS?

Reply:
Changed some line to make the logic clearer.